src/ServiceInsight/Framework/Modules/CoreModule.cs

```diff
@@ -21,7 +21,7 @@ protected override void Load(ContainerBuilder builder)
             builder.RegisterType<HeaderContentDecoder>().As<IContentDecoder<IList<HeaderInfo>>>();
             builder.RegisterType<NetworkOperations>().SingleInstance();
             builder.RegisterType<AppLicenseManager>().SingleInstance();
-            builder.RegisterType<ServiceControlConnectionProvider>().InstancePerLifetimeScope();
+            builder.RegisterType<ServiceControlConnectionProvider>().SingleInstance();
```

Comment:
Turns out we are using lifetime scoping, so InstancePerLifetimeScope is correct, and the real issue was elsewhere.